### PR TITLE
Allow disabled tests to be re-enabled with IGNORE_DISABLED_ISSUES

### PR DESCRIPTION
**Pull request by janeyx99**
Part 1 of fixing https://github.com/pytorch/pytorch/issues/62359

Test plan: 
1. Check out this PR and run `python setup.py install`. 
2. The test we will be running requires CUDA. If you don't have CUDA, you can try this on another device or simply comment out the skipIf statement before the `test_jit_cuda_extension` test in `test_cpp_extensions_jit.py`
3. Run: `IN_CI=1 python test/run_test.py -i test_cpp_extensions_jit -- -k test_jit_cuda_extension` and notice that it should skip. If it doesn't skip, edit test/.pytorch-disabled-tests.json: modify the platforms list of the first issue (61655) to include whatever platform you are on (macos or linux), and just run `python test/test_cpp_extensions_jit.py -v -k test_jit_cuda_extension --import-disabled-tests` to make sure it skips.
4. Now `export PYTORCH_IGNORE_DISABLED_ISSUES=61655` or `export PYTORCH_IGNORE_DISABLED_ISSUES=34952,61655`.
5. `rm test/.pytorch-*` to clear the cached files.
6. Run the same command as in step 5 and note that it SHOULDN'T skip. It should run.

### CODE REVIEW

File: torch/testing/_internal/common_utils.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
 import json
 import __main__  # type: ignore[import]
 import errno
-from typing import cast, Any, Dict, Iterable, Iterator, Optional, Union
+from typing import cast, Any, Dict, List, Iterable, Iterator, Optional, Union
```
Comment (samestep) [inline torch/testing/_internal/common_utils.py:41]: could put this in alphabetical order but it doesn't really matter
```suggestion
from typing import cast, Any, Dict, Iterable, Iterator, List, Optional, Union
```

File: torch/testing/_internal/common_utils.py [hunk 2]
```diff
@@ -426,6 +426,17 @@ def _check_module_exists(name):
...
+# Enables tests disabled by existing issues (these tests are disabled by default)
+# This is usually used locally or during fix PRs. This will enable
+# ALL disabled tests, even if IGNORE_DISABLED_ISSUES is set to a smaller set.
+# This does NOT interfere with other test disablement and only disables tests
+# marked by issues.
+RUN_DISABLED_TESTS = os.getenv('PYTORCH_RUN_DISABLED_TESTS', '0') == '1'
+
+# A list of issue numbers. This will re-enable any disabled tests associated with
+# the issues in this list.
+IGNORE_DISABLED_ISSUES: List[str] = os.getenv('IGNORE_DISABLED_ISSUES', '').split(',')
```
Comment (samestep) [inline torch/testing/_internal/common_utils.py:438]: I'm curious why one of these env vars starts with `PYTORCH_` and the other doesn't

**Review by samestep — APPROVED** [review]
the `IGNORE_DISABLED_ISSUES` part looks good! the `RUN_DISABLED_TESTS` bit seems a bit redundant but I guess it can't hurt

File: torch/testing/_internal/common_utils.py (hunk 1 above)
Comment (janeyx99) [reply torch/testing/_internal/common_utils.py:41]: I ordered them with color: 
image
Is that bad?

**Review by walterddr — CHANGES_REQUESTED** [review]
Hmm. I am not sure this is the best idea/approach...

We currently pass in the DISABLED_TESTS_FILE to common_utils.py so that it parses it. 
and the file is generated (downloaded via tools/testing/)
can we modify the download logic instead and skip them there?

**Comment by janeyx99** [comment]
> Hmm. I am not sure this is the best idea/approach...
> 
> We currently pass in the DISABLED_TESTS_FILE to common_utils.py so that it parses it.
> and the file is generated (downloaded via tools/testing/)
> can we modify the download logic instead and skip them there?

So is the idea there the fact that PYTORCH_IGNORE_DISABLED_ISSUES would only be set once/not reset again? I think switching to the other approach would make switching between different issues for this more difficult. I guess your concern is that this is too downstream/not the right place for this logic, right?

**Review by walterddr — APPROVED** [review]
lgtm

**Pull request closed without merging**